proseco/core.py

```diff
@@ -156,6 +197,96 @@ def __init__(self, data=None, print_log=False, **kwargs):
         for name in ('ra', 'dec', 'RA_PMCORR', 'DEC_PMCORR'):
             self._default_formats[name] = '.6f'
 
+    def set_kwargs(self, **kwargs):
+        for name, val in kwargs.items():
+            if name in self.allowed_kwargs:
+                setattr(self, name, val)
+            else:
+                raise ValueError(f'unexpected keyword argument "{name}"')
+
+        # If an explicit obsid is not provided to all getting parameters via mica
+        # then all other params must be supplied.
+        all_pars = all(getattr(self, x) is not None for x in self.required_attrs)
+        if self.obsid == 0 and not all_pars:
+            raise ValueError('if `obsid` not supplied then all other params are required')
+
+        # If not all params supplied then get via mica for the obsid.
+        if not all_pars:
+            from mica.starcheck import get_starcheck_catalog
+            self.log(f'getting starcheck catalog for obsid {self.obsid}')
+
+            obs = get_starcheck_catalog(self.obsid)
+            obso = obs['obs']
+
+            if self.att is None:
+                self.att = [obso['point_ra'], obso['point_dec'], obso['point_roll']]
+            if self.date is None:
+                self.date = obso['mp_starcat_time']
+            if self.t_ccd is None:
+                self.t_ccd = obs.get('pred_temp', -15.0)
```

Comment:
How does this now get assigned for acq and guide?  Or does it not need to be set separately in this paradigm?

Reply:
Wait.  Found it below so just deleting both comments.